### DISCUSSION TITLE
How to change the prometheus scrapeInterval

### DISCUSSION THREAD

**Discussion by betonetotbo** [prometheus-operator/kube-prometheus#1548] [Q&A] [answered]
How can I change the default `scrapeInterval: 30s` to something like `scrapeInterval: 10s` ?

**Comment by betonetotbo** [comment, marked as answer]
I found a way to do this after applying the manifests, with this command:

`kubectl -n monitoring patch prometheus/k8s --type=json --patch='[{"op":"replace","path":"/spec/scrapeInterval","value":"15s"}]'`

This will patch the [PrometheusSpec](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#prometheusspec) object setting the `scrapeInterval` to `15s`. After a while the `prometheus-operator` will apply this new configuration to the prometheus instance.

**Reply by paulfantom** [reply]
You can also apply this jsonnet patch:
```
prometheus+: {
  prometheus+: {
    spec+: {
      scrapeInterval: "15s",
    },
  },
},
```

**Reply by betonetotbo** [reply]
Worked!

Just for curiosity, I was trying to add it inside the `values`, but it'll not work. It must be a neighbor of `values`:

```javascript
local kp =
  (import 'kube-prometheus/main.libsonnet') +
  {
    values+:: {
      common+: {
        namespace: 'monitoring',
      },
      // here will not work!
    },
    // here it works!
  };
```

**Reply by paulfantom** [reply]
Correct. Currently, only a subset of Prometheus CRD options are supported in `values`, but all are available by directly modifying `prometheus.prometheus.spec`. We are working on bringing this support in `values` in https://github.com/prometheus-operator/kube-prometheus/pull/1477
